packages/untp-playground/src/components/CredentialUploader.tsx

```diff
@@ -1,15 +1,31 @@
 'use client';
 
 import { Card } from '@/components/ui/card';
+import { useError } from '@/contexts/ErrorContext';
 import { jwtDecode } from 'jwt-decode';
 import { useCallback } from 'react';
 import { useDropzone } from 'react-dropzone';
 import { toast } from 'sonner';
 
-export function CredentialUploader({ onCredentialUpload }: { onCredentialUpload: (credential: any) => void }) {
+export function CredentialUploader({
+  onCredentialUpload,
+  setFileCount,
+}: {
+  onCredentialUpload: (credential: any) => void;
+  setFileCount: (count: number) => void;
+}) {
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const { resetErrors } = useError();
+
   const onDrop = useCallback(
     (acceptedFiles: File[]) => {
+      resetErrors();
       const validExtensions = ['.json', '.jwt', '.txt'];
+      if (acceptedFiles.length > 1) {
+        setFileCount(acceptedFiles.length);
+      } else {
+        setFileCount(1);
+      }
```

Comment:
What is logically different between the above 5 lines and:
```suggestion
      setFileCount(acceptedFiles.length);
```